lib/graphql/query/serial_execution/value_resolution.rb

```diff
@@ -45,7 +45,8 @@ class ListResolution < BaseResolution
           def non_null_result
             wrapped_type = field_type.of_type
             strategy_class = get_strategy_for_kind(wrapped_type.kind)
-            value.map do |item|
+            value.each_with_index.map do |item, index|
+              irep_node.index = index
```

Comment:
> Maybe one good thing would be to put it back to nil when we're done, just in case?

Reseting `index` back to `nil` in adf0c0febf44be3e501b32cee04566f428a7df5a